### PR TITLE
Src/add controllers implements

### PR DESCRIPTION
**Pull request by Amet015**

### CODE REVIEW

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 1]
```diff
@@ -29,12 +36,16 @@
 public class ImageConverterController {
 
     // Constant upload file
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Add a point at the last

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 2]
```diff
@@ -29,12 +36,16 @@
...
-    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resources/";
+    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resource/";
+
+    // Constant path converted file.
+    private static final String CONVERTED_FILE = "src/main/java/com/jalasoft/jfc/resource/";
+    // * @param inputPathFile contains the input path of the image.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: It belongs here?

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 3]
```diff
@@ -29,12 +36,16 @@
...
 
     /**
      * imageConverter method receives an image to convert.
      * @param file contains the image file.
-     * @param inputPathFile contains the input path of the image.
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Delete this code line

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 4]
```diff
@@ -45,18 +56,60 @@
...
+        Param param = new ImageParam();
+        ImageParam imageParam = (ImageParam) param;
+        IConverter imageConverter = new ImageConverter();
         try {
             byte[] bytes = file.getBytes();
             Path path = Paths.get(UPLOADED_FOLDER + file.getOriginalFilename());
             Files.write(path, bytes);
+            imageParam.setInputPathFile(path.toString());
         } catch (IOException e) {
             throw new ConvertException("To Do Message","To Do Method where it was generated");
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Is it correct?

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 5]
```diff
@@ -45,18 +56,60 @@
...
+        imageParam.setWidthOfFile(widthOfFile);
+        imageParam.setHeightOfFile(heightOfFile);
+        imageParam.setWhiteBlankPercentage(whiteBlankPercentage);
+        imageParam.setDegreesToRotate(degreesToRotate);
+
+        imageConverter.convert(imageParam);
+
+        return "convert" + imageConverter.convert(imageParam);
+    }
+    private ImageFormat selectFormatImage(String formatImage) {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java:89]: Add a code line here

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 6]
```diff
@@ -45,18 +56,60 @@
...
+        ImageFormat formatImageSelected = null;
+        try{
+            if (formatImage == null){
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Review indentation here please

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 7]
```diff
@@ -45,18 +56,60 @@
...
+                if (formatImage.equals("gif")) {
+                    formatImageSelected = ImageFormat.GIF;
+                }
+                if (formatImage.equals("png")) {
+                    formatImageSelected = ImageFormat.PNG;
+                }
+                if (formatImage.equals("jpg") || formatImage.equals("jpeg") ) {
+                    formatImageSelected = ImageFormat.JPEG ;
+                }
+                if(formatImage.equals("tif")){
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Review indentation here please.

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 8]
```diff
@@ -45,18 +56,60 @@
...
+                    formatImageSelected = ImageFormat.TIFF;
+                }
+            }
+        }catch (NullPointerException e){
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: Review indentation here please.

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java [hunk 9]
```diff
@@ -39,10 +39,10 @@
...
     // Constant upload file.
-    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resources/";
+    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resource/";
 
     // Constant path converted file.
-    private static final String CONVERTED_FILE = "src/main/java/com/jalasoft/jfc/resources/";
+    private static final String CONVERTED_FILE = "src/main/java/com/jalasoft/jfc/resource/";
 
     /**
      * pdfConverter method receives a PDF to convert.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: Here the name method is not necessary

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java [hunk 10]
```diff
@@ -51,20 +51,20 @@
...
     @PostMapping
     public String pdfConverter(
             @RequestParam("file") MultipartFile file, @RequestParam (defaultValue = CONVERTED_FILE)
-            String outputPathFile, @RequestParam String outputFileName, @RequestParam(defaultValue = "0") int rotate,
-            @RequestParam (defaultValue = "1") String scale, @RequestParam (defaultValue = "100") int dpi,
-            @RequestParam String imageType, @RequestParam String formatImage) throws ConvertException {
+            String outputPathFile, @RequestParam String outputFileName, @RequestParam(defaultValue = "90") int rotate,
+            @RequestParam (defaultValue = "300%") String scale, @RequestParam (defaultValue = "x128")String thumbnail,
+            @RequestParam (defaultValue = ".png")String imageFormat, @RequestParam (defaultValue = "1024") int wight,
+            @RequestParam (defaultValue = "720") int height, @RequestParam String pagesToConvert) throws ConvertException {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: Review indentation here please.

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java [hunk 11]
```diff
@@ -76,53 +76,9 @@ public String pdfConverter(
...
-                }
-            }
-        }catch (IllegalArgumentException e){
-            throw new ConvertException("To Do Message","To Do Method where it was generated");
-        }
-        return imageTypeSelected;
+        return "convert " + pdfConverter.convert(pdfParam);
     }
 }
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: Delete a code line here

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 12]
```diff
@@ -33,13 +38,14 @@
...
 
     // Constant upload file.
-    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resources/";
+    private static final String UPLOADED_FOLDER = "src/main/java/com/jalasoft/jfc/resource/";
+
+    // Constant path converted file.
+    private static final String CONVERTED_FILE = "src/main/java/com/jalasoft/jfc/resource/";
 
     /**
      * videoConverter method receives an video to convert
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: The name method is not necessary

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 13]
```diff
@@ -58,20 +64,46 @@
      */
     @PostMapping
     public String videoConverter(
-            @RequestParam("file") MultipartFile file, @RequestParam String fFmpeg, @RequestParam String inputPathFile,
-            @RequestParam String outputPathFile, @RequestParam String outputFileName, @RequestParam double aspectRatio,
-            @RequestParam String frameRate, @RequestParam int width, @RequestParam int height,
-            @RequestParam String videoCodec, @RequestParam String audioCodec, @RequestParam String videoBitRate,
-            @RequestParam String audioBitRate, @RequestParam byte quality, @RequestParam byte channelsNumber,
-            @RequestParam String volume, @RequestParam String rotate) throws ConvertException {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Review this code line 67

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 14]
```diff
@@ -58,20 +64,46 @@
...
+            double aspectRatio, @RequestParam (defaultValue = "30") String frameRate,
+            @RequestParam (defaultValue = "800") int width, @RequestParam (defaultValue = "600")int height ,
+            @RequestParam (defaultValue = "") String videoCodec, @RequestParam (defaultValue = "") String audioCodec,
+            @RequestParam (defaultValue = "") String videoBitRate, @RequestParam (defaultValue = "")
+                    String audioBitRate, @RequestParam (defaultValue = "-1") int quality,
+            @RequestParam (defaultValue = "0") int channelsNumber,
+            @RequestParam (defaultValue = "") String volume,
+            @RequestParam (defaultValue = "") String rotate)  throws ConvertException {
+
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java:77]: Delete a code line here

File: src/main/resources/application.properties [hunk 15]
```diff
@@ -0,0 +1,4 @@
+# max file size
+spring.http.multipart.max-file-size=10MB
+# max request size
+spring.http.multipart.max-request-size=10MB
```
Comment (Enrique-C) [inline src/main/resources/application.properties]: Add a code line at the last

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java (hunk 9 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: ok, it is already fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 8 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 7 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 6 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 5 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java:89]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 4 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 3 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 2 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java (hunk 1 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java (hunk 10 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java (hunk 11 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: ok, it is alreadye fixed

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java (hunk 12 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: ok, it is already fixed

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java [hunk 16]
```diff
@@ -30,9 +30,8 @@
 /**
  * Manage PdfConverter Requests.
  *
- * @version 0.1 14 Dic 2019.
- *
  * @author Enrique Carrizales
+ * @version 0.1 14 Dic 2019.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java:35]: Add a code line after 33

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 17]
```diff
@@ -29,9 +29,8 @@
 /**
  * Manage VideoConverter Requests.
  *
- * @version 0.1 13 Dic 2019.
- *
  * @author Enrique Carrizales
+ * @version 0.1 13 Dic 2019.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java:34]: Add a code line after 32

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 18]
```diff
@@ -44,38 +43,38 @@
...
+     * @param videoCodec     contains videoCodec value.
+     * @param audioCodec     contains audioCodec value.
+     * @param videoBitRate   contains videoBitRate value.
+     * @param audioBitRate   contains audioBitRate value.
+     * @param quality        contains quality of video.
      * @param channelsNumber contains number of output channels.
-     * @param volume contains the level of sound.
-     * @param rotate degrees of rotation.
+     * @param volume         contains the level of sound.
+     * @param rotate         degrees of rotation.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Review the indentation here please

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 19]
```diff
@@ -24,54 +29,79 @@
...
+        VideoParam videoParam = (VideoParam) param;
+        IConverter videoConverter = new VideoConverter();
         try {
             byte[] bytes = file.getBytes();
             Path path = Paths.get(UPLOADED_FOLDER + file.getOriginalFilename());
             Files.write(path, bytes);
+            videoParam.setInputPathFile(path.toString());
         } catch (IOException e) {
-            throw new ConvertException("To Do Message","To Do Method where it was generated");
+            throw new ConvertException("To Do Message", "To Do Method where it was generated");
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Review this code line please

**Review by Enrique-C — APPROVED** [review]

**Pull request merged**